tests/servers/test_navigation_server_2d.h

```diff
@@ -0,0 +1,47 @@
+/**************************************************************************/
+/*  test_navigation_server_2d.h                                           */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_NAVIGATION_SERVER_2D_H
+#define TEST_NAVIGATION_SERVER_2D_H
+
+#include "servers/navigation_server_2d.h"
+
+#include "tests/test_macros.h"
+
+namespace TestNavigationServer2D {
+TEST_SUITE("[Navigation]") {
+	TEST_CASE("[NavigationServer2D] Server should be empty when initialized") {
+		NavigationServer2D *navigation_server = NavigationServer2D::get_singleton();
+		CHECK_EQ(navigation_server->get_maps().size(), 0);
```

Comment:
Also check that regions, links, and agents are all empty?

Reply:
There is no public API for that. That would be only doable in the context of the existing map if we'd have one.

Reply:
ah, nevermind, those are map specific.